Doc/library/socket.rst

```diff
@@ -161,10 +161,25 @@ created.  Socket addresses are represented as follows:
 
   .. versionadded:: 3.7
 
-- Certain other address families (:const:`AF_PACKET`, :const:`AF_CAN`)
-  support specific representations.
-
-  .. XXX document them!
+- The :const:`AF_PACKET` address family is a Linux-only packet address protocol
+  represented by the tuple ``(ifname, proto[, pkttype[, hatype[, addr]]])``
```

Comment:
I haven’t heard of “packet address protocols” before. What does it mean, and why did you add it?

Reply:
The man page for [socket](http://man7.org/linux/man-pages/man2/socket.2.html) has:
```
       Name                Purpose                          Man page
       AF_UNIX, AF_LOCAL   Local communication              unix(7)
       AF_INET             IPv4 Internet protocols          ip(7)
       AF_INET6            IPv6 Internet protocols          ipv6(7)
       AF_IPX              IPX - Novell protocols
       AF_NETLINK          Kernel user interface device     netlink(7)
       AF_X25              ITU-T X.25 / ISO-8208 protocol   x25(7)
       AF_AX25             Amateur radio AX.25 protocol
       AF_ATMPVC           Access to raw ATM PVCs
       AF_APPLETALK        AppleTalk                        ddp(7)
       AF_PACKET           Low level packet interface       packet(7)
       AF_ALG              Interface to kernel crypto API
```
I think I should have said `Linux-only packet interface` or `Linux-only low level packet interface`.  I'm not sure why I changed that to protocol.  I may have read it in a description somewhere, but I can't find the source now.

Reply:
I think the important thing here is to describe that it's a low-level interface directly to network devices.

Reply:
Done.